tests/dummy/app/templates/docs/glimmer-components.md

```diff
@@ -0,0 +1,109 @@
+# Glimmer Components
+
+<aside>
+
+If you’re entirely unfamiliar with Glimmer components, we recommend [the Ember Guides][guides-component] to understand that first!
+
+</aside>
+
+
+Glimmer Components are defined in one of three ways: with templates only, with a template and a backing class, or with only a backing class (i.e. a `yield`-only component). When using a backing class, you get a first-class experience using TypeScript! Unfortunately, we don’t yet support type-checking for templates, but we hope to build that out eventually. Don’t let that stop you, though: types in your component classes make for a great experience, so let’s dig in and see how it works in practice.
+
+## A simple component
+
+A *very* simple Glimmer component which lets you change the count of a value might look like this:
+
+```htmlbars
+<button {{on "click" this.minus}}>&minus;</button>
+{{this.count}}
+<button {{on "click" this.plus}}>+</button>
+```
+
+```ts
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking'
+import { action } from '@ember/object';
+
+export default class Counter extends Component {
+  @tracked count = 0;
+  
+  @action
+  plus() {
+    this.count += 1;
+  }
+  
+  @action
+  minus() {
+    this.count -= 1;
+  }
+}
+```
+
+Notice that there are no type declarations here – but this *is* actually a well-typed component. The type of `count` is `number`, and if we accidentally wrote something like `this.count = null` the compiler would give us an error.
```

Comment:
> if we accidentally wrote something like `this.count = null` the compiler would give us an error.

only if `strictNullChecks` is turned on

Reply:
Yeah, fair. I'll add a parenthetical somewhere explaining what the assumptions of the docs are. `strictNullChecks: true` is our *default*, so people have to explicitly opt out of it, and I'd like to keep these docs focused on the main path (while filling out answers and edge cases and so on in ancillary docs)—I don't think we should be trying to teach *TypeScript* with them, though given our audience I could see an argument that way. Along those lines, we might want a minimal TS primer that explains our configs and why we've chosen them. I'll add it to the project.

Reply:
Do we want this documentation to only apply for apps that preserve our generated config, or is there some benefit to making it config-agnostic where possible?

Reply:
Tradeoffs abound. I think we can probably solve this particular one by using an example that works in both settings (assign a string instead, for example). Maybe we make good use of that kind of example and then to *also* document our norms and why, and call it out if any particular point in the docs depends on our configs.